internal/httpclient/policy_logging_test.go

```diff
@@ -0,0 +1,231 @@
+package httpclient
```

Comment:
For all the test around logging can we pass a logger that log into a buffer and verify the expected content ? 
These tests just verify there is no error but it should tests the log line output as expected. 
You should be able to reuse NewWithOutput from logging package no ?